db.js

```diff
@@ -1,40 +1,42 @@
-const { Pool } = require("pg");
+const knex = require('knex');
 const { loadSecrets } = require("./aws/ssm-helper")
 const config = require("./config.json");
 require("dotenv").config();
 
 let db_username, db_host, db_name, db_password, db_port;
-let pool = null;
+let knexInstance = null;
 
 async function initDb() {
-  const isLocal = process.env.ENVIRONMENT === "local";
-  if (isLocal) {
-    db_username = process.env.DB_USER;
-    db_host = process.env.DB_HOST || "localhost";
-    db_name = process.env.DB_NAME;
-    db_password = process.env.DB_PASS;
-    db_port = process.env.DB_PORT || 5432;
-  } else {
-    // For remote RSD access, load secrets from AWS SSM
-    const awsConfig = config.aws;
-    const credentials = await loadSecrets(awsConfig.region, ['/catbytes_webplatform/db_username', '/catbytes_webplatform/db_password'], true);
-    
-    db_username = credentials['db_username']; 
-    db_host = awsConfig.db_endpoint;
-    db_name = awsConfig.db_name;
-    db_password = credentials['db_password'];
-    db_port = awsConfig.db_port;
-  }
+    const isLocal = process.env.ENVIRONMENT === "local";
+    if (isLocal) {
+        db_username = process.env.DB_USER;
+        db_host = process.env.DB_HOST || "localhost";
+        db_name = process.env.DB_NAME;
+        db_password = process.env.DB_PASS;
+        db_port = process.env.DB_PORT || 5432;
+    }
+    else {
+        // For remote RSD access, load secrets from AWS SSM
+        const awsConfig = config.aws;
+        const credentials = await loadSecrets(awsConfig.region, ['/catbytes_webplatform/db_username', '/catbytes_webplatform/db_password'], true);
 
-  pool = new Pool({
-    user: db_username,
-    host: db_host,
-    database: db_name,
-    password: db_password,
-    port: db_port,
-    connectionTimeoutMillis : 5000,
-    ssl: isLocal ? false : { rejectUnauthorized: false }
-  });
+        db_username = credentials['db_username'];
+        db_host = awsConfig.db_endpoint;
+        db_name = awsConfig.db_name;
+        db_password = credentials['db_password'];
+        db_port = awsConfig.db_port;
+    }
+    knexInstance = knex({
+        client: 'pg',
+        connection: {
+            host: db_host,
+            user: db_username,
+            password: db_password,
+            database: db_name,
+            port: db_port
+        },
+        connectionTimeoutMillis: 5000,
+        ssl: isLocal ? false : { rejectUnauthorized: false }
```

Comment:
did you try running this code? :)
access to RDS still doesn't work

please move these two lines inside the `connection` object

Reply:
no it's fine, it just was't in the right place, now it works, thank you! 🙌🏻